src/NUnitEngine/nunit.engine/RuntimeFramework.cs

```diff
@@ -511,7 +414,26 @@ private static string GetDefaultDisplayName(RuntimeType runtime, Version version
         {
             string displayName;
 
-            if (version == DefaultVersion)
+            Type monoRuntimeType = Type.GetType("Mono.Runtime", false);
+            MethodInfo monoGetDisplayNameMethod = monoRuntimeType?.GetMethod(
+                "GetDisplayName", BindingFlags.Static | BindingFlags.NonPublic | BindingFlags.DeclaredOnly | BindingFlags.ExactBinding);
+
+            if (runtime == RuntimeType.Mono && monoGetDisplayNameMethod != null)
+            {
+                displayName = (string)monoGetDisplayNameMethod.Invoke(null, new object[0]);
+
+                int space = displayName.IndexOf(' ');
+                if (space >= 3) // Minimum length of a version
+                {
+                    string monoVersion = displayName.Substring(0, space);
+                    displayName = "Mono " + version;
+                    if (MonoVersion == null)
+                        MonoVersion = new Version(monoVersion);
+                }
+                else
+                    displayName = "Mono " + displayName;
+            }
```

Comment:
Mono display names were previously set differently depending if it was a CurrentFramework or AvailableFramework. I moved this functionality here mainly for accessibility reasons, although it would be good to also resolve this inconsitency.